### PR TITLE
Refactor structure and parts interactions

### PR DESCRIPTION
**Pull request by ulian-onua**
@memstel this PR contains refactored version of Annotations framework.

The key concepts are:

- The new approach separates logic and rendering. The rendering is mostly generic, based on layers, styles and views. DrawableView doesn’t know anything about exact models;

- The models structure in Models Managers is homogenous with the usage of protocol AnnotationModels. The actions under models (move, resize, change color etc) are performed through common methods and applicable to any model. In the previous version, there were multiple arrays that contained models of each type (pens, highlights, rects etc) that complicated things and contained duplicated methods because of that;

- All math calculations are performed on a logic level based on actual annotation models. Mouse touch events are redirected to the logic level and processed there depending on model coordinates data (depending if there is an annotation model under mouse press, knobs positions if need to move an annotation or drag knob etc);

- Based on Combine streams renderer updates rendering on canvas each time the models' data structure is updated. It is not just a blind update of all models but a smarter one that can be used for some specific cases (for example, when a user edits text annotation only some UI parts of it should be updated);

- The new approach to the History of performed actions that are based on the wrapper of the system’s NSUndoManager and closures. The new history manager is returned by the framework. History is managed by the Annotations framework now but returned as a public instance. Now the history of the annotation canvas can be combined with other actions inside the app that uses it (for example, on Zappy pinned window). It is possible to add multiple actions to a single History event and undo / redo it later;

- No more bunch of delegate callbacks that must have been implemented as empty methods but weren’t used anymore. Now with Combine flow things are more lightweight and easier;

- Text Annotations are completely integrated with the new flow. Before that, it was inside a separate framework for some time and then was added through adapters. It required some duplications to use text annotations together with other annotations. Now text annotations are just a type of annotation model and all common approaches are applied to it. It made it possible to reduce a lot of code;

- A new approach made it possible to reduce a lot of code. The same flow is used for rendering any type of annotations which makes it easier to debug and improve; 

- A new JSON serialisation / deserialisation approach.

https://user-images.githubusercontent.com/15073398/197354059-4ef209be-ef16-4653-bcff-94575aa0e805.mp4

image

### CODE REVIEW

File: Example/Annotations/Playground Controller/PlaygroundViewController.swift [hunk 1]
```diff
@@ -1,84 +1,239 @@
...
 
-class PlaygroundViewController: NSViewController {
+enum LocalError: LocalizedError {
+  case cantLoadFromBundle(_ name: String, _ ext: String)
   
-  var loadViewClosure: ((PlaygroundViewController) -> Void)?
+  var errorDescription: String? {
+    switch self {
+    case .cantLoadFromBundle(let name, let ext):
+      return "Can't create url from bundle from \(name).\(ext)"
```
Comment (memstel) [inline Example/Annotations/Playground Controller/PlaygroundViewController.swift:12]: move static texts to localizable files

File: Example/Annotations/Views/CanvasControlsView.swift [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+      return "Rect"
+    case .obfuscate:
+      return "Obfuscate"
+    case .highlight:
+      return "Highlight"
+    case .number:
+      return "Number"
+    }
+  }
+}
```
Comment (memstel) [inline Example/Annotations/Views/CanvasControlsView.swift:35]: move static texts to localizable files

File: Sources/Annotations/Classes/Managers/Analytics.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      }
+    case is Pen:
+      return "Pen"
+    case is Number:
+      return "Number"
+    case is Text:
+      return "Text"
+    default:
+      return "Unknown"
+    }
```
Comment (memstel) [inline Sources/Annotations/Classes/Managers/Analytics.swift]: move static texts to localizable files

**Review by memstel — CHANGES_REQUESTED** [review]

File: Example/Annotations/Views/CanvasControlsView.swift (hunk 2 above)
Comment (ulian-onua) [reply Example/Annotations/Views/CanvasControlsView.swift:35]: @memstel these strings are in the Example project so I don't think it is something that needs to be localized.

File: Example/Annotations/Playground Controller/PlaygroundViewController.swift (hunk 1 above)
Comment (ulian-onua) [reply Example/Annotations/Playground Controller/PlaygroundViewController.swift:12]: @memstel these strings are in the Example project so I don't think it is something that needs to be localized.

File: Sources/Annotations/Classes/Managers/Analytics.swift (hunk 3 above)
Comment (ulian-onua) [reply Sources/Annotations/Classes/Managers/Analytics.swift]: @memstel done. Moved to Localizable.strings of Package.

**Review by memstel — APPROVED** [review]

**Pull request merged**